src/main/kotlin/com/autonomousapps/model/Coordinates.kt

```diff
@@ -14,16 +14,16 @@ sealed class Coordinates(
    */
   override fun compareTo(other: Coordinates): Int {
     return if (this is ProjectCoordinates) {
-      if (other !is ProjectCoordinates) 1 else identifier.compareTo(other.identifier)
-    } else if (this is ModuleCoordinates) {
+      if (other is ProjectCoordinates) identifier.compareTo(other.identifier) else 1
+    } else if (this is FlatCoordinates) {
+      if (other is ProjectCoordinates) -1 else if (other is FlatCoordinates) gav().compareTo(other.gav()) else 1
+    } else {
       when (other) {
         is ProjectCoordinates -> -1
-        is FlatCoordinates -> 1
+        is FlatCoordinates -> -1
         is ModuleCoordinates -> identifier.compareTo(other.identifier)
         is IncludedBuildCoordinates -> identifier.compareTo(other.identifier)
       }
-    } else {
-      if (other !is FlatCoordinates) 1 else gav().compareTo(other.gav())
```

Comment:
Moved things around a bit to make it easier to read. The test now makes sure that the overall thing does what it should.